linkerd/main/src/main/scala/io/buoyant/linkerd/Main.scala

```diff
@@ -34,12 +34,14 @@ object Main extends App {
         val admin = initAdmin(config, linker)
         val telemeters = linker.telemeters.map(_.run())
         val routers = linker.routers.map(initRouter(_))
+        val closableNamers = linker.namers.map(_._2).collect { case x: Closable => x }
 
         log.info("linkerd initialized.")
         registerTerminationSignalHandler(config.admin.flatMap(_.shutdownGraceMs))
         closeOnExit(Closable.sequence(
           Closable.all(routers: _*),
           Closable.all(telemeters: _*),
+          Closable.all(closableNamers: _*),
```

Comment:
It's not a problem currently because the curator announcer close is synchronous (value future). 
I changed it anyways to support waiting for futures.